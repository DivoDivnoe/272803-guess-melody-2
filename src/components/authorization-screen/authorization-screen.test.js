import React from 'react';
import renderer from 'react-test-renderer';
import AuthorizationScreen from './authorization-screen.jsx';

describe(`AuthorizationScreen component`, () => {
  it(`is rendered correctly`, () => {
    const mock = {
      name: ``,
      password: ``,
      onChange: jest.fn(),
      onSubmit: jest.fn()
    };
    const {name, password, onChange, onSubmit} = mock;

    const tree = renderer.create(
        <AuthorizationScreen
          name={name}
          password={password}
          onChange={onChange}
          onSubmit={onSubmit}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

