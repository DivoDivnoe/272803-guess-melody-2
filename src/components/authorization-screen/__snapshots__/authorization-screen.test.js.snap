// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthorizationScreen component is rendered correctly 1`] = `
<section
  className="login"
>
  <div
    className="login__logo"
  >
    <img
      alt="Угадай мелодию"
      height="83"
      src="img/melody-logo.png"
      width="186"
    />
  </div>
  <h2
    className="login__title"
  >
    Вы настоящий меломан!
  </h2>
  <p
    className="login__total"
  >
    За 0 минут и 30 секунд вы набрали 18 баллов (10 быстрых), совершив 0 ошибок
  </p>
  <p
    className="login__text"
  >
    Хотите сравнить свой результат с предыдущими попытками? Представтесь!
  </p>
  <form
    action="#"
    className="login__form"
    onSubmit={[Function]}
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <p
      className="login__field"
    >
      <label
        className="login__label"
        htmlFor="name"
      >
        Логин
      </label>
      <input
        className="login__input"
        id="name"
        name="name"
        onChange={[Function]}
        type="text"
        value=""
      />
    </p>
    <p
      className="login__field"
    >
      <label
        className="login__label"
        htmlFor="password"
      >
        Пароль
      </label>
      <input
        className="login__input"
        id="password"
        name="password"
        onChange={[Function]}
        type="text"
        value=""
      />
      <span
        className="login__error"
      >
        Неверный пароль
      </span>
    </p>
    <button
      className="login__button button"
      disabled={true}
      type="submit"
    >
      Войти
    </button>
  </form>
</section>
`;
